Allow pecl like usage in ext/pdo

tianon opened the pull request:
This is similar to what was discussed/fixed over in #2955, although with a slightly simpler solution.

IMO this is cleaner because it makes the command's arguments match the exact string being tested by `make` (and could even use `$@` / `$<` if we wanted to get fancy).

Not that it's terribly important, but https://github.com/docker-library/php/issues/618 is the reason this caught my attention (which is on PHP 7.0 and turns out isn't necessary since PDO is included in the image already, but still seemed like a nice thing to send a PR upstream for :heart:).

I'm happy to adjust, rebase, amend, re-target, etc as desired!

nikic commented:
As mentioned in https://github.com/php/php-src/pull/2955#issuecomment-352189187, the reason why it is currently implemented this way is to avoid including absolute paths in the generate line number annotations. As the generated lexers are committed to the repo, these environment-specific paths will cause unnecessary diffs (and potentially leak information). As far as I can see, this fix also suffers from the same problem, right?

Maybe a better solution would be to cd to `$(srcdir)` rather than `$(top_srcdir)` and drop the `ext/pdo` prefixes?

tianon commented:
@nikic doh, thanks for pointing that out :face_with_head_bandage:

Fixed in 38aa417 to `cd $(srcdir)` instead (and dropped the subshell since every line in a `Makefile` runs in a new shell instance). :+1:

weltling commented:
@tianon IMO it is better to keep the php root relative path, when it comes to checking in into the repo. For the generation under docker it doesn't matter much probably. But that line info can be actually used for debug.

Thanks.

tianon commented:
@weltling do you have a concrete alternative proposal that accomplishes that and still fixes the bug?

weltling commented:
@tianon i'd suggest same solution as in the PR you mention in the description, please check https://github.com/php/php-src/commit/c5768a7e2441204bce30d24d867bd2b3e8618213. I'm not sure it makes sense to compile ext/pdo shared, btw. Otherwise, if the Docker on-the-fly process doesn't affect the generated code in the repo, seems acceptable.

Thanks.

tianon commented:
Ok, fair enough -- updated in 0f3148519a873f3d37961cedfe77eac401ade1a1 to match the implementation in c5768a7e2441204bce30d24d867bd2b3e8618213. :+1:
